src/rrt/Tree.hpp

```diff
@@ -363,7 +363,9 @@ class Tree {
 
         //  Make sure there's actually a direct path from @source to
         //  @intermediateState.  If not, abort
-        if (!_stateSpace->transitionValid(source->state(), intermediateState)) {
+        if (!_stateSpace->transitionValid(
+                _forward ? source->state() : intermediateState,
+                _forward ? intermediateState : source->state())) {
```

Comment:
This might be slightly cleaner if you declare two temporary variables right before this to use in the if statement. I'm ok either way though.

```
const T& from = _forward ? source->state() : intermediateState;
const T& to = _forward ? intermediateState : source->state();
if (!_stateSpace->transitionValid(from, to) {
```